cache/src/main/java/net/runelite/cache/definitions/WorldMapCompositeDefinition.java

```diff
@@ -36,4 +38,35 @@ public class WorldMapCompositeDefinition
 	public final Set<MapSquareDefinition> mapSquareDefinitions = new HashSet<>();
 	public final Set<ZoneDefinition> zoneDefinitions = new HashSet<>();
 	public final List<WorldMapElementDefinition> worldMapElementDefinitions = new ArrayList<>();
+
+	public Position calculateWorldOffset(Position position)
+	{
+		int squareX = position.getX() / 64;
+		int squareZ = position.getY() / 64;
+		int zoneX = (int)(((position.getX() / 64.0) - squareX) * 8.0);

```

Comment:
Is this something you copied from the client code? It mathematically is just `position.getX() & 7`

Reply:
Changed it. Did not copy, was just guessing what 'zone' meant and came up with that

Reply:
Looking at this again, I believe they are different. `1052 & 7` = 4, but `(int)(((1052 / 64.0) - ((int)(1052/64))) * 8.0)` = 3

Reply:
Oh I misread this. It is `(x & 63)/8`. Where is this reference code from you are using?

Reply:
Yep that works, thanks. I don't have a reference for this since, as far as I know, there are no 'source' positions for this stuff as it was created for the world map which is laid out differently for multi level stuff. This is just an effort to figure out where this stuff is located for the source squares/zones.